feat:Redifinition de l'inscription client

Kouvahe81 opened the pull request:
Cette fonctionnalité  est la même que celle définie dans feature/US7-2/Signup-create-user

Martin-Hayot commented:
hello serge, je me suis permis de tester et faire des ajouts sur ta feature qui marche très bien. je n'ai rien touché niveau fonctionnelle juste rajouté des éléments qui sont un plus pour cet us.

voici les modifications : 
- Ajout d'un bouton pour le signup avec google
- ajout d'un lien pour changer de page entre login et signup
- retrait de la navbar qui n'est pas necessaire qui va devenir un composant
- retrait de la couleur blanche du background car elle doit changer en fonction du darkmode ou non

Martin-advinci reviewed (approved):
Parfait !

Kouvahe81 committed:
US7-2 Feat: Page inscription
client.
Cette fonctionnalité permet de créer un utilisateur via un formulaire.
Différents test sont fait à savoir:

la validité du mot de passe
la vérification du mot de passe et celle de confirmation
la force du mot de passe

Martin-Hayot committed:
fix: double import de Link

Martin-Hayot committed:
refactor: background color dynamique avec darkmode

Julien00859 commented on g-sonde/src/pages/Signup.tsx:
```diff
@@ -0,0 +1,276 @@
+        }
+    };
+    async function getGoogleOAuthUrl() {
+        const res = await axios.get("http://localhost:3001/api/auth/google");
```
L'URL pour le retour de l'oauth semble hardcodée sur localhost, quid de si le serveur et le client sont sur des machines séparées ? comme dans une situation classique où le client serait venu via un nom de domaine

Julien00859 commented on g-sonde/src/pages/Signup.tsx:
```diff
@@ -0,0 +1,276 @@
+            return;
+        }
+        axios
+            .post("http://localhost:3001/api/auth/signup", {
```
Encore une URL hardcodée ici

Julien00859 commented on g-sonde/src/pages/Signup.tsx:
```diff
@@ -0,0 +1,276 @@
+        const mediumRegex = new RegExp(
+            "^(((?=.*[a-z])(?=.*[A-Z]))|((?=.*[a-z])(?=.*[0-9]))|((?=.*[A-Z])(?=.*[0-9])))(?=.{6,})"
+        );
+        if (strongRegex.test(value)) {
+            setBackgroundColor("#0F9D58");
+        } else if (mediumRegex.test(value)) {
+            setBackgroundColor("#F4B400");
+        } else {
+            setBackgroundColor("#DB4437");
+        }
```
ça aurait été bien de mettre ces valeurs hexadécimale dans des variables nommées du nom de la couleur, ou bien à défaut un commentaire, ici les couleurs sont donc dans l'ordre: vert, orange, rouge

Julien00859 commented on g-sonde/src/pages/Signup.tsx:
```diff
@@ -0,0 +1,276 @@
+
+    // Pour tester la force du mot de passe à travers les codes couleurs
+    const analyze = (value: string) => {
+        const strongRegex = new RegExp(
+            "^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#_\\$%^&*])(?=.{8,})"
+        );
+        const mediumRegex = new RegExp(
+            "^(((?=.*[a-z])(?=.*[A-Z]))|((?=.*[a-z])(?=.*[0-9]))|((?=.*[A-Z])(?=.*[0-9])))(?=.{6,})"
+        );
```
Intuitivement je comprend ces deux regexps comme:
strong => au moins 1 minuscule, au moins 1 majuscule, au moins 1 chiffre, au moins 1 spécial, au moins 8 caractères
medium => au moins 2 classes de caractères et au moins 6 caractères
si ce n'est pas ça, il faudrait mettre un peu de documentation
aussi je suis plutôt bon en regexp dans un peu de doc ne ferait pas de mal aux autres (à minima dans le message de commit)

aussi, si on s'intéresse aux maths combinatoires derrière votre politique de mot de passe: 8 caractères, 4 classes de caractères présentes. On obtient en fait le même degré de sécurité que si on n'autorisait *que* les minuscules mais qu'on demandait un minimum de 10 caractères. J'aimerais donc bien vous suggérer d'également regarder la longueur du mot de passe et s'il dépasse 10 qu'il apparaisse vert aussi :) Moi qui utilise un password manager qui me génère des mdp à 20 caractères mais parfois avec que des minuscules, j'apparait en rouge dans votre formulaire alors que je devrais être super vert :D

Julien00859 reviewed (commented):
Le message de commit de l'US7-2 est trop léger, on y lit qu'il y a une validation de la "force" du mot de passe sans plus d'informations, j'aurais voulu lire dans le message à propos de votre politique de mots de passes et aussi savoir si elle était ou non contraignante, càd si on peut ou non s'enregistrer avec un mot de passe "rouge".

J'aurais aussi voulu lire dans le message de commit un rappel de la stratégie dans le backend pour la sauvegarde des mots de passe et de la manière dont la requête s'effectue, ça aurait pu être aussi simple que:

> Le formulaire est toujours envoyé en HTTPS et le mot de passe est hashé avec \<algo\> dans le backend, voir US-X

Sinon j'aime bien le découpage en commits :+1:

Beau boulot pour avoir réussi à remettre de l'ordre dans cet historique et pour avoir réussi à merger cette PR, ci-dessous quelques points d'attentions qui m'ont sautés aux yeux lors de ma propre review. Je comprend les circonstances qui font que cette PR devait être merge au plus vite et je pense que peut-être revenir sur ces points dans une nouvelle PR (si vous trouvez ça adéquat bien entendu) serait une bonne idée.
